### PR TITLE
fix(bot): fix some issue in bot plugin

### PR DESCRIPTION
**Pull request by hund030**
1. Need deploy does not work right.
2. Log message should be with component name.
3. Reuse v1 error handling and telemetry helper.

E2E TEST: no need

### CODE REVIEW

File: packages/fx-core/src/common/azure-hosting/azureHosting.ts [hunk 1]
```diff
@@ -88,11 +88,9 @@ export abstract class AzureHosting {
     module = AzureHosting.replacePluginId(module, pluginId);
 
     return {
-      Configuration: this.configurable
-        ? {
-            Modules: { [this.hostType]: module },
-          }
-        : undefined,
+      Configuration: {
```
Comment (IvanJobs) [inline packages/fx-core/src/common/azure-hosting/azureHosting.ts:91]: Any explanations for :
1. What is `this.configurable` for?
2. Why removing this flag?

**Review by eriolchan — DISMISSED** [review]

File: packages/fx-core/src/common/azure-hosting/azureHosting.ts (hunk 1 above)
Comment (hund030) [reply packages/fx-core/src/common/azure-hosting/azureHosting.ts:91]: If the hosting is not configurable, it does not need to update bicep template.
https://github.com/OfficeDev/TeamsFx/blob/eec8ff4f80d64001d5438cb2a805ac35e0500cb8/packages/fx-core/src/common/azure-hosting/azureHosting.ts#L81

File: packages/fx-core/src/plugins/resource/bot/index.ts [hunk 2]
```diff
@@ -265,90 +252,6 @@ export class TeamsBot implements Plugin {
     );
   }
 
-  private static wrapError(
```
Comment (hund030) [inline packages/fx-core/src/plugins/resource/bot/index.ts:268]: Move this method to error.ts

File: packages/fx-core/src/plugins/resource/bot/index.ts [hunk 3]
```diff
@@ -265,90 +252,6 @@ export class TeamsBot implements Plugin {
...
-              getLocalizedString("plugins.bot.UnhandledError", errorMsg),
-            ],
-            isPluginError(e) ? e.innerError : undefined
-          )
-        );
-      return ResultFactory.SystemError(UnhandledErrorCode, [errorMsg, errorMsg], innerError);
-    }
-  }
-
-  private static async runWithExceptionCatching(
```
Comment (hund030) [inline packages/fx-core/src/plugins/resource/bot/index.ts:335]: Move this method to error.ts

**Review by eriolchan — APPROVED** [review]

**Pull request merged**